torch/fx/experimental/proxy_tensor.py

```diff
@@ -331,15 +331,16 @@ def can_handle_tensor(x):
         return r
 
     # For pre-autograd tracing, we do not want to run CompositeImplicit decomps.
-    if not pre_dispatch and func not in [
+    if func not in [
         torch.ops.aten.size.default,
         torch.ops.aten.stride.default,
         torch.ops.aten.storage_offset.default,
     ]:
         with proxy_mode:
-            r = func.decompose(*args, **kwargs)
-            if r is not NotImplemented:
-                return r
+            if not proxy_mode.export_inference:
```

Comment:
cc @ezyang / @zou3519, any thoughts on another flag gunking up ProxyMode?

The use case is that Tugsuu wants to avoid decomposing CompositeImplicit ops in the inference code path for export. Which requires auditing all the places we normally decompose:

(1) autograd key (CompositeImplicits)
(2) FunctionalTensorMode
(3) ProxyMode

The alternative we discussed was having a layer that converts all composite implicit ops into custom ops during tracing, then a graph pass that converts back after. The main annoyance with that approach though is we'd need to add infra to interpose and actually turn all composite implicits into custom ops during tracing (probably in pre-dispatch somewhere).

Reply:
I'd prefer the flag be named after what it does, rather than who is using it, but being able to impede func.decompose calls seems like a reasonable knob to have.

Reply:
It seems the custom op approach won't be as easy because since we are doing it at pre-dispatch level, we need to make it work with subclass modes in the future.

Reply:
+1 for Ed's comment. A knob on func.decompose seems reasonable and the name should reflect that.

One alternative to threading the flag around these modes is to have it as a single config option somewhere